pkg/machine/os/ostree.go

```diff
@@ -0,0 +1,132 @@
+//go:build amd64 || arm64
+// +build amd64 arm64
+
+package os
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/containers/image/v5/transports/alltransports"
+)
+
+// OSTree deals with operations on ostree based os's
+type OSTree struct { //nolint:revive
+}
+
+// Apply takes an OCI image and does an rpm-ostree rebase on the image
+// If no containers-transport is specified,
+// apply will first check if the image exists locally, then default to pulling.
+// Exec-ing out to rpm-ostree rebase requires sudo, so this means apply cannot
+// be called within podman's user namespace if run as rootless.
+// This means that we need to export images in containers-storage to oci-dirs
+// We also need to do this via an exec, since if we tried to use the ABI functions,
+// we would enter the user namespace, the rebase command would fail.
+func (dist *OSTree) Apply(image string, opts ApplyOptions) error {
+	imageWithTransport := image
+
+	transport := alltransports.TransportFromImageName(image)
+
+	switch {
+	// no transport was specified
+	case transport == nil:
+		exists, err := execPodmanImageExists(image)
+		if err != nil {
+			return err
+		}
+
+		if exists {
+			fmt.Println("Pulling from", "containers-storage"+":", imageWithTransport)
+			dir, err := os.MkdirTemp("", pathSafeString(imageWithTransport))
+			if err != nil {
+				return err
+			}
+			err = os.Chmod(dir, 0755)
+			if err != nil {
+				return err
+			}
+			defer os.RemoveAll(dir)
```

Comment:
in this case, it might be prudent to put it in a go func such that we can log the error.  if this does not make sense, let me know and I will dig out an example.